sdk/core/core/inc/az_json_data.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_JSON_DATA_H
+#define _az_JSON_DATA_H
+
+#include <az_json_token.h>
+#include <az_mut_span.h>
+#include <az_span.h>
+
+#include <stdbool.h>
+
+#include <_az_cfg_prefix.h>
+
+typedef struct az_json_data az_json_data;
+
+typedef struct {
+  struct {
+    az_json_data const * begin;
+    size_t size;
+  } _detail;
+} az_json_array;
+
+AZ_NODISCARD AZ_INLINE size_t az_json_array_size(az_json_array const a) { return a._detail.size; }
+
+typedef struct az_json_object_member az_json_object_member;
+
+typedef struct {
+  struct {
+    az_json_object_member const * begin;
+    size_t size;
+  } _detail;
+} az_json_object;
+
+AZ_NODISCARD AZ_INLINE size_t az_json_object_size(az_json_object const a) { return a._detail.size; }
+
+typedef enum {
+  AZ_JSON_DATA_NULL = 0,
+  AZ_JSON_DATA_BOOLEAN = 1,
+  AZ_JSON_DATA_NUMBER = 2,
+  AZ_JSON_DATA_STRING = 3,
+  AZ_JSON_DATA_OBJECT = 4,
+  AZ_JSON_DATA_ARRAY = 5,
+} az_json_data_kind;
+
+struct az_json_data {
+  az_json_data_kind kind;
+  union {
+    bool boolean;
+    az_span string;
+    double number;
+    az_json_array array;
+    az_json_object object;
+  } data;
+};
+
+typedef struct {
+  bool has;
+  az_json_data data;
+} az_json_data_option;
```

Comment:
Did we agree on option, or optional during the meeting?

Reply:
I don't remember.

Reply:
https://msit.microsoftstream.com/video/b2d0e3b1-3055-4b0b-a8c3-10461ce2686c Around 25:30, we discuss `az_optional_bool` name and `is_present` insteadof `has`. Currently I need to reboot to listen to audio, so I am not sure (auto-subtitles are not very helpful)